src/integrationTest/java/uk/gov/hmcts/reform/cpo/utils/CasePaymentOrderAuditUtils.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.cpo.utils;
+
+import org.hibernate.envers.AuditReader;
+import org.hibernate.envers.AuditReaderFactory;
+import org.hibernate.envers.DefaultRevisionEntity;
+import org.hibernate.envers.RevisionType;
+import org.hibernate.envers.query.AuditEntity;
+import org.hibernate.envers.query.AuditQuery;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.cpo.data.CasePaymentOrderEntity;
+import uk.gov.hmcts.reform.cpo.domain.CasePaymentOrderAuditRevision;
+
+import javax.persistence.EntityManagerFactory;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+public class CasePaymentOrderAuditUtils {
+
+    @Autowired
+    private EntityManagerFactory entityManagerFactory;
+
+    private final AuditReader auditReader;
+
+    public static final boolean SELECT_DELETED_ENTITIES = true;
+    public static final boolean SELECT_ENTITIES_ONLY = true;
+    public static final boolean SELECT_REVISION_INFO_ONLY = false;
+
+    public CasePaymentOrderAuditUtils(EntityManagerFactory entityManagerFactory) {
+        this.auditReader = AuditReaderFactory.get(entityManagerFactory.createEntityManager());
+    }
+
+    public AuditQuery getAuditQuery(UUID id, boolean shouldSelectEntitiesOnly) {
+        return auditReader
+            .createQuery()
+            .forRevisionsOfEntity(
+                CasePaymentOrderEntity.class,
+                shouldSelectEntitiesOnly,
+                SELECT_DELETED_ENTITIES)
+            .add(AuditEntity.id().eq(id));
+    }
+
+    public List<CasePaymentOrderAuditRevision> getAuditRevisions(UUID id) {
+        @SuppressWarnings("unchecked")
+        List<Object[]> resultList = getAuditQuery(id, SELECT_REVISION_INFO_ONLY).getResultList();
+
+        return resultList.stream()
+            .map(array -> CasePaymentOrderAuditRevision
```

Comment:
Make it look so simple and elegant, I made a real ugly mess of doing the same thing

Reply:
I would not have been able to get there without yor CPO-16 and CPO-8 code.  Thanks